### PR TITLE
* #54: Added check from an HTTP response from the Github when trigger…

### LINKED ISSUE #54: Check the HTTP response code after triggering the GitHub workflow
## Situation

Currently, we only send the request and report back that the release is successful. 
As a user, I want to see an error in case of workflows fails to start

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #54

### CODE REVIEW

File: src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java [hunk 1]
```diff
@@ -81,14 +81,21 @@ public void sendGitHubRequest(final URI uri, final String json) throws GitHubExc
...
+            validateResponse(response);
         } catch (final IOException | InterruptedException exception) {
             Thread.currentThread().interrupt();
             throw new GitHubException("F-RR-GH-2: Exception happened during sending an HTTP request to the GitHub.",
                     exception);
         }
     }
 
+    private void validateResponse(final HttpResponse<String> response) throws GitHubException {
+        if (response.statusCode() < 200 || response.statusCode() >= 300) {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java:94]: isn't enough to check if the status is not 200?

`if (HttpURLConnection.HTTP_OK != response.statusCode()) { `

**Review by chiaradiamarcelo — APPROVED** [review]
looks good!

File: src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java:94]: GitHub sends me a 204 response, not 200. Maybe it's safer to have a range?

Comment (chiaradiamarcelo) [reply src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java:94]: According to https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_success , Http successful codes go from 200 to 208, so you could use that range.

**Review by chiaradiamarcelo — DISMISSED** [review]

File: src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java (hunk 1 above)
Comment (chiaradiamarcelo) [reply src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java:94]: Actually, by reading the doc, it looks like Github will always send a 204 https://docs.github.com/en/free-pro-team@latest/rest/reference/repos

So better check only for that

**Pull request merged**